olive/passes/onnx/gptq.py

```diff
@@ -0,0 +1,170 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import logging
+from typing import Any, Dict
+
+import torch
+from packaging import version
+
+from olive.hardware.accelerator import AcceleratorSpec, Device
+from olive.model import PyTorchModelHandler
+from olive.model.handler.onnx import ONNXModelHandler
+from olive.passes import Pass
+from olive.passes.pass_config import PassConfigParam
+
+logger = logging.getLogger(__name__)
+
+
+class GptqQuantizer(Pass):
+    """GPTQ quantization using Hugging Face Optimum and export model with onnxruntime optimized kernel."""
+
+    @staticmethod
+    def _default_config(accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigParam]:
+        return {
+            "nsamples": PassConfigParam(
+                type_=int,
+                default_value=128,
+                description="number of samples in calibration dataset to apply quantization. Default value is 128",
+            ),
+            "bits": PassConfigParam(
+                type_=int,
+                default_value=4,
+                description="quantization bits. Default value is 4",
+            ),
+            "dataset": PassConfigParam(
```

Comment:
do we need unify with data_config? @trajepl ?

Reply:
do we need update the type like [https://github.com/microsoft/Olive/blob/2374ab9b5d8a70829667f9ada09a94c7a3be76c2/olive/passes/pytorch/lora.py#L194](https://github.com/microsoft/Olive/blob/2374ab9b5d8a70829667f9ada09a94c7a3be76c2/olive/passes/pytorch/lora.py#L194)